### DISCUSSION TITLE
Custom struct properties stripped

### DISCUSSION THREAD

**Discussion by itsthesteve** [pocketbase/pocketbase#5575] [Q&A]
Thank you for pocketbase, I've learned a lot using this project.

I have a quiz style app that pulls data from a `terms` table with the defined columns (term, definition, etc). For certain types of terms I would like to add another field to the struct (and the final JSON output) but it seems anything I add is nullified in the JSON response. 

I've read a few discussions and PRs about this but I can't find anything that solves this particular issue.

The following model has everything but `FooBar` defined in the collection schema.

```go

var _ models.Model = (*Terms)(nil)

type Terms struct {
	models.BaseModel

	StartsWith string `json:"startsWith"`
	Term       string `json:"term"`
	Definition string `json:"definition"`
	Category   string `json:"category"`
	FooBar     string `json:"foobar"`
}

// This doesn't seem to have any effect
func (t *Terms) TableName() string {
	return "terms"
}

func (t *Terms) DoAnything() {
	t.FooBar = "this doesn't output"
}

```

Inside the `DoAnything` function, I want to populate  a property on the struct based on data inside the defined columns.

Here's the query:

```go
...
result := []m.Terms{}
err := app.Dao().DB().NewQuery(`
	SELECT * FROM terms
	WHERE category = {:category}
	ORDER BY RANDOM() LIMIT {:limit}`).Bind(dbx.Params{
	"category": "abbreviation",
	"limit":    limit,
}).All(&result)

if err != nil {
	fmt.Println(err)
	return nil, ErrDbSelect
}

if len(result) >= MIN_DATASET_COUNT {
	for _, r := range result {
		r.DoAnything()
	}
	return result, nil
}
```

Building the final result and sending:
```go
for range quizSize {
	var group Group
	group.Choice = rand.Intn(limit)
	terms, err := getDefinitions(app, limit, category) // Sampled code from above

	if err != nil {
		fmt.Println("Error getting definitions", err, terms)
		return echo.NewHTTPError(500, err.Error())
	}

	log.Printf("\n%+v\n", terms) // FooBar is nil here too

	group.Terms = append(group.Terms, terms...)
	result.Groups = append(result.Groups, group)
}

return c.JSON(200, result)
```

When the result is output, the key is there, but the value is stripped:

```json
{
  "id": "rc97df2183b5fa6",
  "created": "2024-09-26 21:51:36.508Z",
  "updated": "2024-09-26 21:51:36.508Z",
  "startsWith": "I",
  "term": "IR",
  "definition": "Infrared",
  "category": "abbreviation",
  "foobar": ""
}
```

I saw some discussions on custom marshaling, but haven't had much luck.

Maybe it's a go thing or maybe it's a pb thing, so far all attempts have failed.

Thanks!

**Comment by ganigeorgiev** [comment]
When executing `r.DoAnything()` it seems that you are executing it on a copy of the model.

Try changing  `result := []m.Terms{}` to `result := []*m.Terms{}` OR modify the slice element using its index like `result[idx].DoAnything()`.

**Comment by itsthesteve** [comment]
I was afraid it was a skill issue... thanks Gani!
